diff --git a/setup.py b/setup.py
index 25222ea..faba0c2 100644
--- a/setup.py
+++ b/setup.py
@@ -44,40 +44,14 @@ class CustomSDist(sdist):
 
 
 def build_native(spec):
-    cmd = ["cargo", "build"]
-    if not DEBUG_BUILD:
-        cmd.append("--release")
-        target = "release"
-    else:
-        target = "debug"
-
-    # Step 0: find rust sources
-    if not os.path.isfile("../cabi/Cargo.toml"):
-        scratchpad = tempfile.mkdtemp()
-
-        @atexit.register
-        def delete_scratchpad():
-            try:
-                shutil.rmtree(scratchpad)
-            except (IOError, OSError):
-                pass
-
-        zf = zipfile.ZipFile("rustsrc.zip")
-        zf.extractall(scratchpad)
-        rust_path = scratchpad + "/rustsrc/cabi"
-    else:
-        rust_path = "../cabi"
-        scratchpad = None
-
+    target = "release"
     # Step 1: build the rust library
-    build = spec.add_external_build(cmd=cmd, path=rust_path)
+    build = spec.add_external_build(cmd="true", path="./rust/")
 
     rtld_flags = ["NOW"]
-    if sys.platform == "darwin":
-        rtld_flags.append("NODELETE")
     spec.add_cffi_module(
         module_path="semaphore._lowlevel",
-        dylib=lambda: build.find_dylib("semaphore", in_path="target/%s" % target),
+        dylib=lambda: build.find_dylib("semaphore", in_path="target/release"),
         header_filename=lambda: build.find_header("semaphore.h", in_path="include"),
         rtld_flags=rtld_flags,
     )
